### ISSUE TITLE
Reopen: escape pk/fk property names in 4.x.x querydsl

### ISSUE THREAD

**Issue by dmitrymurashenkov** [querydsl/querydsl#2251]
Ran into #861 using querydsl-maven-plugin 4.1.4 to generate code from SQL schema with Oracle.

Some of my PKs have name equal to table name and generated classes have both PrimaryKey field and static field for entity generated with same name. 

Was #861 fix not merged into 4.x branch?

**Comment by dmitrymurashenkov** [comment]
Update - in my case the conflict is between default variable name and primary key name. Since primary key property name uniqueness is only checked against other properties - the conflict still possible.

```
public class EntitySerializer implements Serializer {
    ...
    protected void intro(EntityType model, SerializerConfig config,
                CodeWriter writer) throws IOException {
        ...
        if (config.createDefaultVariable()) {
            introDefaultInstance(writer, model, config.defaultVariableName());
        }
```

```
public class DefaultNamingStrategy extends AbstractNamingStrategy {
    ...
    @Override
    public String getPropertyNameForPrimaryKey(String pkName, EntityType entityType) {
        ...
        if (!entityType.getEscapedPropertyNames().contains(candidate)) {
            return normalizeJavaName(candidate);
        }
    }
```

**Comment by dmitrymurashenkov** [comment]
Seems there can be a workaround by providing customized SerializerConfig into GenericExporter with custom defaultVariableName or createDefaultVariable=false.

**Comment by Shredder121** [comment]
*Huh* I also can't reach the server it's hosted on.
Will have to wait on Timo, to see if he can sign into the dashboard.

As for the uniqueness of the variable name, that sounds like an issue to pick up indeed.
